sdk/azcore/runtime/pipeline.go

```diff
@@ -23,7 +24,20 @@ type PipelineOptions struct {
 
 // Pipeline represents a primitive for sending HTTP requests and receiving responses.
 // Its behavior can be extended by specifying policies during construction.
-type Pipeline = exported.Pipeline
+type Pipeline struct {
+	internal  exported.Pipeline
+	traceProv tracing.Provider
+}
+
+// Do implements the policy.Transporter interface for the Pipeline type.
+func (p Pipeline) Do(req *policy.Request) (*http.Response, error) {
+	return p.internal.Do(req)
+}
+
+// TracingProvider returns the tracing provider associated with this Pipeline.
+func (p Pipeline) TracingProvider() tracing.Provider {
```

Comment:
Not sure how much I like this.  While it's convenient and solves the issue, the tracing provider doesn't really have anything to do with the pipeline at a conceptual level.

Reply:
In C++, the tracing provider is in the Client options. The ctor code looks for it being set and then adds the policy if necessary to the pipeline. The provider itself is another field in the client. So, the pipeline doesn't know anything about tracing - just another policy gets injected into it.

Reply:
How will the client name get passed to [`CreateTracer`](https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/inc/azure/core/tracing/tracing.hpp#L31)?